aws-lc-rs/src/ec/signature.rs

```diff
@@ -192,43 +193,86 @@ impl VerificationAlgorithm for EcdsaVerificationAlgorithm {
         public_key: &[u8],
         msg: &[u8],
         signature: &[u8],
+    ) -> Result<(), Unspecified> {
+        let public_key = parse_ec_public_key(public_key, self.id.nid())?;
+        self.verify_ecdsa(msg, signature, &public_key)
+    }
+
+    fn verify_digest_sig(
+        &self,
+        public_key: &[u8],
+        digest: &digest::Digest,
+        signature: &[u8],
+    ) -> Result<(), Unspecified> {
+        let public_key = parse_ec_public_key(public_key, self.id.nid())?;
+
+        self.verify_digest_ecdsa(digest, signature, &public_key)
+    }
+}
+
+impl EcdsaVerificationAlgorithm {
+    fn verify_ecdsa(
+        &self,
+        msg: &[u8],
+        signature: &[u8],
+        public_key: &LcPtr<EVP_PKEY>,
     ) -> Result<(), Unspecified> {
         match self.sig_format {
             EcdsaSignatureFormat::ASN1 => {
-                verify_asn1_signature(self.id, self.digest, public_key, msg, signature)
+                verify_asn1_signature(self.digest, public_key, msg, signature)
             }
             EcdsaSignatureFormat::Fixed => {
                 let (out_bytes, out_bytes_len) = convert_fixed_signature(self.id, signature)?;
-                verify_asn1_signature(self.id, self.digest, public_key, msg, unsafe {
+                verify_asn1_signature(self.digest, public_key, msg, unsafe {
                     out_bytes.as_slice(out_bytes_len)
                 })
             }
         }
     }
 
-    fn verify_digest_sig(
+    fn verify_digest_ecdsa(
         &self,
-        public_key: &[u8],
-        digest: &digest::Digest,
+        digest: &Digest,
         signature: &[u8],
+        public_key: &LcPtr<EVP_PKEY>,
     ) -> Result<(), Unspecified> {
         if self.digest != digest.algorithm() {
             return Err(Unspecified);
         }
         match self.sig_format {
             EcdsaSignatureFormat::ASN1 => {
-                verify_asn1_digest_signature(self.id, digest, public_key, signature)
+                verify_asn1_digest_signature(digest, public_key, signature)
             }
             EcdsaSignatureFormat::Fixed => {
                 let (out_bytes, out_bytes_len) = convert_fixed_signature(self.id, signature)?;
-                verify_asn1_digest_signature(self.id, digest, public_key, unsafe {
+                verify_asn1_digest_signature(digest, public_key, unsafe {
                     out_bytes.as_slice(out_bytes_len)
                 })
             }
         }
     }
 }
 
+impl ParsedVerificationAlgorithm for EcdsaVerificationAlgorithm {
+    fn parsed_verify_sig(
+        &self,
+        public_key: &ParsedPublicKey,
+        msg: &[u8],
+        signature: &[u8],
+    ) -> Result<(), Unspecified> {
+        self.verify_ecdsa(msg, signature, public_key.key())
```

Comment:
Should this and all the other implementations check that the passed in public_key type / algorithm matches? Right now this appears to rely on the AWS-LC APIs to catch the failure for the public key type being incorrect from the looks of it.